src/rpp/sockets.cpp

```diff
@@ -1958,6 +1993,48 @@ namespace rpp
         return get_ip_interface(network_interface, af).broadcast.str();
     }
 
+    uint64_t get_network_handle(const std::string& network_interface) noexcept
+    {
+#if __ANDROID__ && __ANDROID_API__ >= 21
+        using namespace rpp::jni;
+
+        Class Activity{"android/app/Activity"};
+        Class ConnectivityManager{"android/net/ConnectivityManager"};
+        Class LinkProperties{"android/net/LinkProperties"};
+        Class Network{"android/net/Network"};
+
+        Method getSystemService = Activity.Method("getSystemService", "(Ljava/lang/String;)Ljava/lang/Object;");
+        Method getAllNetworks = ConnectivityManager.Method("getAllNetworks", "()[Landroid/net/Network;");
+        Method getLinkProperties = ConnectivityManager.Method("getLinkProperties", "(Landroid/net/Network;)Landroid/net/LinkProperties;");
+        Method getInterfaceName = LinkProperties.Method("getInterfaceName", "()Ljava/lang/String;");
+        Method getNetworkHandle = Network.Method("getNetworkHandle", "()J");
+
+        Ref<jobject> connectivityManager = getSystemService.Object(getActivity(), String("connectivity").get());
+
+        Ref<jobjectArray> networks = getAllNetworks.Array(connectivityManager);
+        jsize length = getEnv()->GetArrayLength(networks.get());
+
+        for (jsize i = 0; i < length; ++i)
+        {
+            Ref<jobject> network{getEnv()->GetObjectArrayElement(networks, i)};
+            Ref<jobject> linkProperties = getLinkProperties.Object(connectivityManager, network);
+
+            if (linkProperties)
+            {
+                std::string interfaceName = getInterfaceName.String(linkProperties);
```

Comment:
Doesn't `String()` return a JString? Which requires `.str()` to be called. (I changed this yesterday)